src/embeddings.rs

```diff
@@ -247,6 +274,15 @@ impl PyEmbeddings {
                 .map_err(|err| exceptions::IOError::py_err(err.to_string())),
         }
     }
+
+    fn iter_with_norm(slf: PyRefMut<Self>) -> PyResult<PyObject> {
+        let gil = Python::acquire_gil();
```

Comment:
This method could be simplified, I'm not sure whether my suggestion has further implications, but analogous to the `vocab()` method, we could use:

```Rust
    fn iter_with_norm(&self) -> PyResult<PyEmbeddingWithNormIterator> {
        Ok(PyEmbeddingWithNormIterator::new(self.embeddings.clone(), 0))
    }
```